plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1368,12 +1368,23 @@ public VM createVmFromTemplate(final Connection conn, final VirtualMachineTO vmS
 
         final String bootArgs = vmSpec.getBootArgs();
         if (bootArgs != null && bootArgs.length() > 0) {
+            // send boot args for PV instances
             String pvargs = vm.getPVArgs(conn);
             pvargs = pvargs + vmSpec.getBootArgs().replaceAll(" ", "%");
             if (s_logger.isDebugEnabled()) {
                 s_logger.debug("PV args are " + pvargs);
             }
             vm.setPVArgs(conn, pvargs);
+
+            // send boot args into xenstore-data for HVM instances
+            Map<String, String> xenstoreData = new HashMap<>();
+
+            xenstoreData.put("vm-data/cloudstack/init", bootArgs);
+            vm.setXenstoreData(conn, xenstoreData);
+
+            if (s_logger.isDebugEnabled()) {
+                s_logger.debug("HVM args are " + bootArgs);
```

Comment:
Is this if statement still required to check if debug is enabled?

Reply:
Technically it can be removed, but as its javadoc says: `This function is intended to lessen the computational cost of disabled log debug statements.` and the fact that they are used all over the place I added the same if block.

Reply:
In this type of system such as ACS, I believe it is worth the cost of a tiny bit of more processing to remove an If statement. Specially in cases such as this one, when you already have others IFs where this one is nested.

Reply:
I didn't get your point! Are your pro on this specific IF block or against it?

Reply:
What I meant is that, in a Web API drive system such as ACS, everything is synchronous (or almost everything). Therefore, the 1 nanoseconds of improvement that the use of this IF conditional can bring is negligible. This means that the code (the code used to create this optimization) is not worth it (in my perspective).

I normally see people using this approach of checking log level before logging when creating log messages using String.format, which uses quite some processing. However, even in those cases, I do not see much need to check the log level before logging (in most cases).

I just commented to give some context on the use of this is<LogLevel>Enabled method. The only time I used this approach of checking log level was when I developed some standalone optimization algorithms that had a time constraint to be executed. Therefore, there was no space to waste clock cycles creating a log message that is not going to be used.

In summary; if I were coding, I would not use it. However, we have these things spread all over our code base. I normally remove them, but if you want to use, the choice is yours.

Reply:
Makes totally sense to me. Removed the IF condition.